exotica_core/src/task_map.cpp

```diff
@@ -73,24 +73,36 @@ void TaskMap::Update(Eigen::VectorXdRefConst x, Eigen::VectorXdRef Phi, Eigen::M
     if (jacobian.rows() != TaskSpaceDim() && jacobian.cols() != x.rows())
         ThrowNamed("Jacobian dimension mismatch!");
 
-    // Store original x (needs to be reset later).
-    Eigen::VectorXd x_original(x);
-    Update(x_original, Phi);
-
-    // Backward finite differencing.
+    // Set constants
     constexpr double h = 1e-6;
-    Eigen::VectorXd x_tmp;
-    Eigen::VectorXd phi_original(Phi);
-    for (int i = 0; i < TaskSpaceDim(); ++i)
+    constexpr double hi = 1.0 / h;
+    int ntask = TaskSpaceDim();
+    int ndof = jacobian.cols();
```

Comment:
`ndof` is only used once, no need to maintain a variable for that. Directly use it in the `for`